### ISSUE TITLE
Allow Django Integration Usernames to be Numbers

### ISSUE THREAD

**Issue by cyface** [getsentry/sentry#67601]
### Problem Statement

Hello!

 Today I am getting a warning when using the Django integration and the "default PII" setting.
 
 Our username fields are numbers, so I get this message on every alert:
 
Discarded invalid value (1)
user.username
expected a string
value: xxxx

I would love a way to tell Sentry that it's OK that our usernames are numbers and not have it discard them.

### Solution Brainstorm

An init param of username_format_validation=False, or username_format=int or something like that.

### Product Area

Ingestion and Filtering

**Comment by jjbayer** [comment]
Sounds like a reasonable ask to me, will discuss with the team before proceeding.

**Comment by jjbayer** [comment]
@cyface do you know how the integer ends up in the username field? Is it the Django integration writing it into the event? That would mean that Django's [`User.get_username()`](https://docs.djangoproject.com/en/5.0/ref/contrib/auth/#django.contrib.auth.models.User.get_username) returns an integer in your case.

https://github.com/getsentry/sentry-python/blob/05d1e5ca94cc4fffcd01c46ceda6713459308404/sentry_sdk/integrations/django/__init__.py#L592

**Comment by cyface** [comment]
Yes that is correct. User.get_username() returns a number in my app. We have Django Auth connected to an SSO system that puts the SSO unique ID in the username field. Thank you!

**Comment by jjbayer** [comment]
Will be fixed by https://github.com/getsentry/relay/pull/3328.
